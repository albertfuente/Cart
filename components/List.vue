<template>
  <div class="list-component">
    <div v-for="item in items" :key="item.id" class="list-component-container">
      <div @click="onClick(item.id)" class="list-component-motorbike">
        <Item :label="item.id" />
      </div>
    </div>
  </div>
</template>

<script>
import Item from "../atoms/Item.vue";

export default {
  data() {
    return {
      items: [],
      selected: 1,
    };
  },
  components: {
    Item,
  },
  methods: {
    async fetch() {
      const result = await this.$axios.get(`https://fake.dev.mapit.me/motos`);
      this.items = result.data;
    },
    onClick(id) {
      window.location.href = `motorbike/${id}`;
    },
  },
  mounted() {
    this.fetch();
  },
};
</script>
<style scoped>
.list-component {
  border: 1px solid black;
  margin: 10px;
  padding: 10px;
}
.list-component-container {
  width: 50%;
  background: #0ea7de;
  display: flex;
  flex-direction: column;
  border: 2px solid black;
  justify-content: center;
  margin: 0px auto;
  align-items: center;
}
.list-component-motorbike {
  padding: 10px;
  text-transform: uppercase;
  font-weight: bold;
  border-bottom: 1px solid white;
  width: 95%;
  text-align: center;
  cursor: pointer;
}
</style>