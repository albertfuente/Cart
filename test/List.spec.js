import { shallowMount } from '@vue/test-utils'
import List from './../components/List.vue'

describe('List.vue/List', () => {
  it('mounts the component and creates a snapshot inside the __snapshots__ folder', () => {
    const wrapper = shallowMount(List)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('is rendering main div with correct class', () => {
    const wrapper = shallowMount(List)
    expect(wrapper.classes()).toContain('list-component')
  })

})