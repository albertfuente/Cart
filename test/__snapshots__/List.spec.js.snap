// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue/List mounts the component and creates a snapshot inside the __snapshots__ folder 1`] = `"<div class=\\"list-component\\"></div>"`;
